params/config.go

```diff
@@ -865,3 +1018,36 @@ func (c *ChainConfig) ToWithUpgradesJSON() *ChainConfigWithUpgradesJSON {
 		UpgradeConfig: c.UpgradeConfig,
 	}
 }
+
+// Checks if messages have the same hash
+//
+// `message` is the simulation of a configuration being parsed from the local
+// config. `message2` is parsing a message being exchanged through the network
+// (a foreign config), and `message3` is the the deserialization and
+// serialization of the foreign config. All 3 instances should have the same
+// hashing, depite maybe not being identical (some configurations may be in a
+// different order, but our hashing algorithm is resilient to those changes,
+// thanks for our serialization library, which produces always the same output.
+func AssertConfigHashesAndSerialization(t *testing.T, originalConfig *UpgradeConfig) {
+	bytes, err := originalConfig.MarshalBinary()
+	require.NoError(t, err)
+
+	deserializedConfig := UpgradeConfig{}
+	require.NoError(t, deserializedConfig.UnmarshalBinary(bytes))
+
+	twiceDeserialized := UpgradeConfig{}
+	newBytes, err := deserializedConfig.MarshalBinary()
+	require.NoError(t, err)
+	require.NoError(t, twiceDeserialized.UnmarshalBinary(newBytes))
```

Comment:
When will this fail because we do it three times instead of two?

If `newBytes` is the same as `bytes` (which should be confirmed by checking that the hashes match), then I don't think that `twiceDeserialized` provides additional coverage. Afaict this will fail if `UnmarshalBinary` is itself non-deterministic on the SAME bytes.